File: examples-nosql-java-sdk/sqlexamples/Namespaces.java
/*Copyright (c) 2020, 2023 Oracle and/or its affiliates.
 * Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
*/
import oracle.nosql.driver.NoSQLHandle;
import oracle.nosql.driver.NoSQLHandleConfig;
import oracle.nosql.driver.NoSQLHandleFactory;
import oracle.nosql.driver.kv.StoreAccessTokenProvider;
import oracle.nosql.driver.ops.SystemResult;
import oracle.nosql.driver.ops.SystemRequest;

public class Namespaces{
   /* Name of your namespace */
   final static String nsName = "ns1";

   public static void main(String[] args) throws Exception {
      //Replace the placeholder below with your full hostname
      String kvstore_endpoint ="http://<your_hostname>:8080";
      NoSQLHandle handle = generateNoSQLHandleonPrem(kvstore_endpoint);
      try {
         createNS(handle);
         dropNS(handle);
      } catch (Exception e) {
         System.err.print(e);
      } finally {
         handle.close();
      }
   }
   /* Create a NoSQL handle to access the onPremise Oracle NoSQL database */
   private static NoSQLHandle generateNoSQLHandleonPrem(String kvstore_endpoint) throws Exception {
      NoSQLHandleConfig config = new NoSQLHandleConfig(kvstore_endpoint);
      config.setAuthorizationProvider(new StoreAccessTokenProvider());
      /* If using a secure store pass the username, password of the store to StoreAccessTokenProvider*/
      /*config.setAuthorizationProvider(new StoreAccessTokenProvider(username, password));*/
      NoSQLHandle handle = NoSQLHandleFactory.createNoSQLHandle(config);
      return handle;
   }

   private static void createNS(NoSQLHandle handle) throws Exception {
      String createNSDDL = "CREATE NAMESPACE IF NOT EXISTS ns1";
      SystemRequest sysreq = new SystemRequest();
      sysreq.setStatement(createNSDDL.toCharArray());
      SystemResult sysres = handle.systemRequest​(sysreq);
      sysres.waitForCompletion​(handle, 60000,1000);
      System.out.println("Namespace " + nsName + " is created");
   }
   private static void dropNS(NoSQLHandle handle) throws Exception {
      String dropNSDDL = "DROP NAMESPACE " + nsName;
      SystemRequest sysreq = new SystemRequest();
      sysreq.setStatement(dropNSDDL.toCharArray());
      SystemResult sysres = handle.systemRequest​(sysreq);
      sysres.waitForCompletion​(handle, 60000,1000);
      System.out.println("Namespace " + nsName + " is dropped");
   }
}

File: examples-nosql-python-sdk/sqlexamples/Namespaces.py
# Copyright (c) 2023, 2024 Oracle and/or its affiliates.
# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
import os
from borneo import (Regions, NoSQLHandle, NoSQLHandleConfig, PutRequest,QueryRequest,
                    TableRequest, SystemRequest, GetRequest, TableLimits, State)
from borneo.iam import SignatureProvider
from borneo.kv import StoreAccessTokenProvider

# Given a endpoint, instantiate a connection to the onPremise Oracle NoSQL database
def get_connection_onprem():
   # replace the placeholder with the name of your host
   kvstore_endpoint ='http://<hostname>:8080'
   provider = StoreAccessTokenProvider()
   # If using a secure store, uncomment the line below and pass the username, password of the store to StoreAccessTokenProvider
   # provider = StoreAccessTokenProvider(username, password)
   return NoSQLHandle(NoSQLHandleConfig(kvstore_endpoint, provider))

# Creates a namespace
def create_ns(handle):
   statement = '''CREATE NAMESPACE IF NOT EXISTS ns1'''
   sysreq = SystemRequest().set_statement(statement)
   sys_result = handle.system_request(sysreq)
   sys_result.wait_for_completion(handle, 40000, 3000)
   print('Created namespace: ns1')

# Drop a namespace
def drop_ns(handle):
   statement = '''DROP NAMESPACE ns1'''
   sysreq = SystemRequest().set_statement(statement)
   sys_result = handle.system_request(sysreq)
   sys_result.wait_for_completion(handle, 40000, 3000)
   print('Namespace: ns1 is dropped')

def main():
   handle = get_connection_onprem()
   create_ns(handle)
   drop_ns(handle)
   os._exit(0)

if __name__ == "__main__":
    main()
